src/Report/JsonReport.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Report;
+
+/**
+ * @author Boris Gorbylev <ekho@ekho.name>
+ *
+ * @internal
+ */
+final class JsonReport implements ReportInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getFormat()
+    {
+        return 'json';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generate(ReportConfig $reportConfig)
+    {
+        $jFiles = array();
+
+        foreach ($reportConfig->getChanged() as $file => $fixResult) {
+            $jfile = array('name' => $file);
+
+            if ($reportConfig->shouldAddAppliedFixers()) {
+                $jfile['appliedFixers'] = $fixResult['appliedFixers'];
+            }
+
+            if (!empty($fixResult['diff'])) {
+                $jfile['diff'] = $fixResult['diff'];
+            }
+
+            $jFiles[] = $jfile;
+        }
+
+        $json = array(
+            'files' => $jFiles,
+        );
+
+        if (null !== $reportConfig->getTime()) {
+            $json['time'] = array(
+                'total' => round($reportConfig->getTime() / 1000, 3),
```

Comment:
Also, I don't think "report config" is a good name for the class if it has information like this in it. That information litterally makes it a "report".